experimental/dds/tree/src/test/LogViewer.tests.ts

```diff
@@ -612,14 +612,62 @@ describe('CachingLogViewer', () => {
 			after: arbitraryRevisionView,
 			steps: [],
 		});
-		const after = viewer.getRevisionViewInSession(Number.POSITIVE_INFINITY);
+		const after = viewer.getRevisionViewInMemory(Number.POSITIVE_INFINITY);
 		expect(editsProcessed).deep.equal([false, true]);
 		expect(after).equal(arbitraryRevisionView);
 		log.addLocalEdit(edit3);
-		viewer.getRevisionViewInSession(Number.POSITIVE_INFINITY);
+		viewer.getRevisionViewInMemory(Number.POSITIVE_INFINITY);
 		expect(editsProcessed).deep.equal([false, true, false]);
 	});
 
+	it('caches the oldest in memory revision when edits are evicted', async () => {
+		let editsProcessed = 0;
+		let editsEvicted = 0;
+		let retainedRevision = 0;
+		const editLogSize = 10;
+		const log = new EditLog<ChangeInternal>(undefined, undefined, undefined, editLogSize);
+		const viewer = getCachingLogViewerAssumeAppliedEdits(log, simpleLogBaseView, () => editsProcessed++);
+
+		log.registerEditEvictionHandler((editsToEvict) => {
+			editsEvicted += editsToEvict;
+		});
+
+		viewer.on(CachingLogViewerDiagnosticEvents.RevisionRetained, (revision) => {
+			retainedRevision = revision;
+		});
+
+		// Add enough edits to the log to start evicting
+		for (let i = 0; i < editLogSize; i++) {
+			const edit = newEdit([]);
+			log.addSequencedEdit(edit, { sequenceNumber: i, referenceSequenceNumber: i - 1 });
+			expect(log.getIndexOfId(edit.id)).equals(i);
+		}
+		expect(log.length)
+			.equals(log.numberOfSequencedEdits)
+			.and.equals(editLogSize, 'Only sequenced edits should be present.');
+
+		// Add more edits so that the oldest get evicted
+		for (let i = 0; i < editLogSize; i++) {
+			const edit = newEdit([]);
+			const sequenceNumber = editLogSize + 1;
+			log.addSequencedEdit(edit, {
+				sequenceNumber,
+				referenceSequenceNumber: sequenceNumber - 1,
+				minimumSequenceNumber: sequenceNumber - 1,
+			});
+			expect(log.getIndexOfId(edit.id)).equals(editLogSize + i);
+		}
+
+		// Check to see that the revision was retained
+		expect(editsEvicted).to.equal(editLogSize);
```

Comment:
IMO the `editsEvicted` and `editsProcessed` are already given this test coverage in the edit log tests, so if you agree and would rather remove those assertions and event listeners here I think it'd be appropriate. On the one hand, it doesn't make things worse from a correctness standpoint to test things twice and have overlapping coverage in tests, in fact, there's maybe a very slight benefit in that if somebody deletes a test by accident, coverage won't be lost. But on the other hand, whenever a refactor or API change needs to happen, it's that many more places to change. And we've had to do a lot of that in this repo :(. Just my $0.02; I'll let you decide what to do.

If these asserts _are_ testing something that isn't tested in edit log, I think they should be moved to their own tests up in EditLog.tests.ts. The reason being that you don't need the log viewer to test these things.

Reply:
Agreed, removed.